pupil_src/shared_modules/marker_tracker_3d/user_interface.py

```diff
@@ -298,10 +324,13 @@ def open_window(self):
 
             glfw.glfwMakeContextCurrent(active_window)
 
+        logger.info("3d visualization window is opened")
+
     def close_window(self):
         if self._window:
             glfw.glfwDestroyWindow(self._window)
             self._window = None
+        logger.info("3d visualization window is closed")
 
     def on_resize(self, window, w, h):
         self.trackball.set_window_size(w, h)
```

Comment:
again, I cannot comment on the actual lines.

`on_window_key`: Better to call `close_window()` instead of `on_close`, that makes semantically more sense to me (because you are already handling the event, you don't need to call another handler). On the other hand, you can use `on_close` for the observer for `cleanup`